types/shallowequal/shallowequal-tests.ts

```diff
@@ -1,15 +1,24 @@
-import shallowEqual_require = require('shallowequal');
-import * as shallowEqual_splat from 'shallowequal';
+import shallowEqual = require('shallowequal');
```

Comment:
I don't know what the current state of the art is on the whole `* as` versus `= require` import styles in Typescript. I see you also deleted the `declare namespace`; will this work properly with both import styles?

Reply:
No, it won't work any more, but this is on purpose.

There is a [clear statement](https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default) on how to handle synthetic default imports. It boils down to activating either `allowSyntheticDefaultImports` or `esModuleInterop` compiler options. This is nothing that has to be emulated via type definitions, especially if it won't actually work in some environments.

Reply:
I don't think that statement is relevant; neither of the import styles I mentioned are default imports. They should be more or less interchangeable because they both bind the imported name to the function, though they may compile differently (which is why I asked in the first place). That's why there was a `declare namespace` line originally. But I don't use these typings anymore anyway, so whatever.

Reply:
It's more of a statement of fact - namespace imports cannot be callable or constructable in es6, so TS attempts to enforce that (it was loose before and a namespace merge could trip it up). So an `import = require` is the canonical way to do an interopy import that is just a require call, which can be callable or constructable.

Reply:
I see, thanks!